sklearn/manifold/t_sne.py

```diff
@@ -142,10 +142,12 @@ def _kl_divergence(params, P, degrees_of_freedom, n_samples, n_components,
 
     # Q is a heavy-tailed distribution: Student's t-distribution
     n = pdist(X_embedded, "sqeuclidean")
-    n += 1.
```

Comment:
I have added the test case, and fixed the bug in barnes_hut as well, glad to say that it passes the test while the old code failed like we thought

Marking the pull request as [MRG] again